miner/worker.go

```diff
@@ -381,7 +390,13 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 			commit(false, commitInterruptNewHead)
 
 		case head := <-w.chainHeadCh:
-			headNumber := head.Block.NumberU64()
+			if h, ok := w.engine.(consensus.Handler); ok {
```

Comment:
You don't need to listen to the `chainHeadCh` anymore.  That `case` can be removed.